packages/lodestar/src/network/reqresp/rateTracker.ts

```diff
@@ -0,0 +1,63 @@
+import {MapDef} from "../../util/map";
+
+type RateTrackerOpts = {
+  limit: number;
+  timeoutMs: number;
+};
+
+/**
+ * The generic rate tracker allows up to `limit` objects in a period of time.
+ * This could apply to both request count or block count, for both requests and responses.
+ */
+export class RateTracker {
+  private requestsWithinWindow = 0;
+  private limit: number;
+  private timeoutMs: number;
+  /** Key as time in second and value as object requested */
+  private requests: MapDef<number, number>;
+
+  constructor(opts: RateTrackerOpts, requests = new MapDef<number, number>(() => 0)) {
+    this.limit = opts.limit;
+    this.timeoutMs = opts.timeoutMs;
+    this.requests = requests;
+  }
+
+  requestObjects(objectCount: number): number {
+    if (objectCount <= 0) throw Error("Invalid objectCount " + objectCount);
+    this.prune();
+    if (this.requestsWithinWindow >= this.limit) {
+      return 0;
+    }
+
+    this.requestsWithinWindow += objectCount;
+    const key = Math.floor(Date.now() / 1000);
```

Comment:
Make this 1000 a const BUCKET_SIZE_MS, and justify the value

Reply:
yeah it's just to convert ms into s and it's not likely we'll change it, we use this 1000 constant in `prune()` method as well so it's good to make a separate constant for it